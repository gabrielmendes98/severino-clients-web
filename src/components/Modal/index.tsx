import Dialog, { DialogProps } from '@mui/material/Dialog';
import DialogContent from '@mui/material/DialogContent';

const Modal = ({ open, onClose, children, ...props }: DialogProps) => (
  <Dialog {...props} open={open} onClose={onClose}>
    <DialogContent>{children}</DialogContent>
  </Dialog>
);

export default Modal;
