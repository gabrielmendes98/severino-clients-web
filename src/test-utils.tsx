import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
