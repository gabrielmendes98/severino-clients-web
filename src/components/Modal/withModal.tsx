/* eslint-disable react/display-name */
import {
  ComponentType,
  FunctionComponent,
  useCallback,
  useMemo,
  useState,
} from 'react';
import { DialogProps } from '@mui/material/Dialog';
import Modal from './index';

interface ModalConfig {
  body: FunctionComponent;
  ModalProps?: Omit<DialogProps, 'open'>;
}

export interface InjectedModalProps {
  showModal: (config: ModalConfig) => void;
}

const DefaultBody = () => <></>;

const withModal =
  <T extends object>(Component: ComponentType<T>) =>
  (props: Omit<T, keyof InjectedModalProps>) => {
    const [open, setOpen] = useState(false);
    const [config, setConfig] = useState<ModalConfig>({
      body: DefaultBody,
    });

    const Body = useMemo(() => config.body, [config.body]);

    const handleClose = useCallback(() => setOpen(false), []);

    const showModal = useCallback((config: ModalConfig) => {
      setConfig(config);
      setOpen(true);
    }, []);

    return (
      <>
        <Component {...(props as T)} showModal={showModal} />

        <Modal {...config.ModalProps} open={open} onClose={handleClose}>
          <Body />
        </Modal>
      </>
    );
  };

export default withModal;
