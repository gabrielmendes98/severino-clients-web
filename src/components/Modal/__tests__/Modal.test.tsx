import { render } from 'test-utils';
import Modal from '../index';

it('should match snapshot when opened', () => {
  const { baseElement } = render(
    <Modal open>
      <div>content</div>
    </Modal>,
  );
  expect(baseElement).toMatchSnapshot();
});

it('should match snapshot when closed', () => {
  const { baseElement } = render(
    <Modal open={false}>
      <div>content</div>
    </Modal>,
  );
  expect(baseElement).toMatchSnapshot();
});
