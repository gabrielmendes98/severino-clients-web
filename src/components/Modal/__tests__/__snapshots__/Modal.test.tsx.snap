// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when closed 1`] = `
<body
  style=""
>
  <div>
    <div
      class="Toastify"
    />
  </div>
</body>
`;

exports[`should match snapshot when opened 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="Toastify"
    />
  </div>
  <div
    class="MuiModal-root MuiDialog-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r0:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-nsysyq-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div>
            content
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
